lib/core/src/Cardano/Wallet/Primitive/AddressDiscovery/Sequential.hs

```diff
@@ -394,36 +374,32 @@ nextChangeIndex pool (PendingIxs ixs) =
 -- [BIP-44](https://github.com/bitcoin/bips/blob/master/bip-0044.mediawiki)
 --
 -- Internally, the state keeps track of a few things for us and is it is
--- parameterized by a type @t@ which represents a target backend. This enables
--- the state to be agnostic to the underlying address format.
-data SeqState n = SeqState
-    { internalPool :: !(AddressPool n 'InternalChain)
+-- parameterized by a type @n@ which captures a particular network discrimination.
+-- This enables the state to be agnostic to the underlying address format.
+data SeqState k = SeqState
+    { internalPool :: !(AddressPool 'InternalChain k)
         -- ^ Addresses living on the 'InternalChain'
-    , externalPool :: !(AddressPool n 'ExternalChain)
+    , externalPool :: !(AddressPool 'ExternalChain k)
         -- ^ Addresses living on the 'ExternalChain'
     , pendingChangeIxs :: !PendingIxs
         -- ^ Indexes from the internal pool that have been used in pending
         -- transactions. The list is maintained sorted in descending order
         -- (cf: 'PendingIxs')
     }
-    deriving stock (Generic, Show)
+    deriving stock (Generic)
 
-instance NFData (SeqState n)
+deriving instance (Show (k 'AccountK XPub), Show (KeyFingerprint "payment" k))
+    => Show (SeqState k)
 
-instance Buildable (SeqState n) where
-    build (SeqState intP extP chgs) = "SeqState:\n"
-        <> indentF 4 (build intP)
-        <> indentF 4 (build extP)
-        <> indentF 4 ("Change indexes: " <> indentF 4 chgsF)
-      where
-        chgsF = blockListF' "-" build (pendingIxsToList chgs)
```

Comment:
I've removed 'Buildable' instances as a lazy way to update them for now :grimacing: ... If they were really needed somewhere, I'll actually update them.. otherwise.. bye.